static/app/views/issueDetails/groupFeatureFlags/useGroupFlagDrawerData.tsx

```diff
@@ -0,0 +1,126 @@
+import {useMemo} from 'react';
+
+import {OrderBy, SortBy} from 'sentry/components/events/featureFlags/utils';
+import type {Group} from 'sentry/types/group';
+import useGroupFeatureFlags from 'sentry/views/issueDetails/groupFeatureFlags/useGroupFeatureFlags';
+import {useGroupSuspectFlagScores} from 'sentry/views/issueDetails/groupFeatureFlags/useGroupSuspectFlagScores';
+import type {GroupTag} from 'sentry/views/issueDetails/groupTags/useGroupTags';
+
+interface SuspectGroupTag extends GroupTag {
+  suspect: {
+    baselinePercent: undefined | number;
+    score: undefined | number;
+  };
+}
+
+interface Props {
+  environments: string[];
+  group: Group;
+  orderBy: OrderBy;
+  search: string;
+  sortBy: SortBy;
+}
+
+interface Response {
+  allFlagCount: number;
+  displayFlags: SuspectGroupTag[];
+  isError: boolean;
+  isPending: boolean;
+  refetch: () => void;
+}
+
+export default function useFlagDrawerData({
+  environments,
+  group,
+  orderBy,
+  search,
+  sortBy,
+}: Props): Response {
+  const isSuspectEnabled = sortBy === SortBy.SUSPICION;
+
+  // Fetch the base flag data
+  const {
+    data: groupFlags = [],
+    isError: isFlagsError,
+    isPending: isFlagsPending,
+    refetch: refetchFlags,
+  } = useGroupFeatureFlags({
+    groupId: group.id,
+    environment: environments,
+  });
+
+  // Fetch the suspect data, if we need it for this render
+  const {
+    data: suspectScores,
+    isError: isSuspectError,
+    isPending: isSuspectPending,
+  } = useGroupSuspectFlagScores({
+    groupId: group.id,
+    environment: environments.length ? environments : undefined,
+    enabled: isSuspectEnabled,
+  });
+
+  // Combine the flag and suspect data into SuspectGroupTag objects
+  const suspectFlags = useMemo(() => {
```

Comment:
```suggestion
  const allFlagsWithScores = useMemo(() => {
```
Or something that's not suspectFlags. When I read suspectFlags, I think tthe highlighted, most suspicious flags.

Reply:
renamed